parametrize the preset json file name in the spring boot configuration

renanfranca opened the pull request:
Fix #10553

murdos commented:
What exactly has changed since https://github.com/jhipster/jhipster-lite/pull/10578?

renanfranca commented:
> What exactly has changed since https://github.com/jhipster/jhipster-lite/pull/10578?

I tagged you in the code comments with explanations. Let me know if it was enough 👍

murdos commented:
> I tagged you in the code comments with explanations. Let me know if it was enough 👍

I have not received any notification. Maybe you've started a review session that you have not finished?

renanfranca commented:
> I have not received any notification. Maybe you've started a review session that you have not finished?


@murdos: Maybe I don't know how to start a review properly when I am the PR owner. I'll play around with two GitHub accounts in another repository to master it, hahaha ;)

### Changes since the previous PR #10578

The bug was fixed by adding this parameter:

```java
@Configuration 
@EnableConfigurationProperties(JHipsterPresetFileProperties.class) 
class JHipsterPresetFilePropertiesConfiguration { 
  
  @Bean 
  JHipsterPresetFile jhipsterPresetFile(JHipsterPresetFileProperties presetFileProperties) { 
    return new JHipsterPresetFile(PresetName.from(presetFileProperties.getName())); 
  } 
}
```

This was the previous bugged code:

```java
@Configuration 
@EnableConfigurationProperties(JHipsterPresetFileProperties.class) 
class JHipsterPresetFilePropertiesConfiguration { 
  
  @Bean 
  JHipsterPresetFile jhipsterPresetFile() { 
    return new JHipsterPresetFile(new JHipsterPresetFileProperties().getName()); 
  } 
}
```

I used a value object instead of a string and refactored the code to fit that change:

```java
package tech.jhipster.lite.project.domain.resource; 
  
import tech.jhipster.lite.project.domain.preset.PresetName; 
import tech.jhipster.lite.shared.error.domain.Assert; 
  
public record JHipsterPresetFile(PresetName name) { 
  public JHipsterPresetFile { 
    Assert.notNull("name", name); 
  } 
}
```

pascalgrimaud commented on src/main/java/tech/jhipster/lite/project/infrastructure/secondary/JHipsterPresetFilePropertiesConfiguration.java:
```diff
@@ -0,0 +1,17 @@
+class JHipsterPresetFilePropertiesConfiguration {
+
+  @Bean
+  JHipsterPresetFile jhipsterPresetFile(JHipsterPresetFileProperties presetFileProperties) {
```
in fact, I would use directly @Value in repository, to inject the property, no need to use a Configuration and a Bean

renanfranca commented:
https://github.com/jhipster/jhipster-lite/pull/10590#discussion_r1721449763

@murdos , please WDYT about it? (I tagged here, because I am not sure if you received notification 😅)
